File: src/main/java/ca/yyx/hu/extractor/MediaExtractorInterface.java
package ca.yyx.hu.extractor;

import android.content.Context;
import android.media.MediaCodec;
import android.media.MediaFormat;
import android.net.Uri;

import java.io.IOException;
import java.nio.ByteBuffer;
import java.util.Map;

/**
 * @author algavris
 * @date 30/04/2016.
 */
public interface MediaExtractorInterface {
    int readSampleData(ByteBuffer buffer, int offset);

    void getSampleCryptoInfo(MediaCodec.CryptoInfo cryptoInfo);

    void release();

    void setDataSource(byte[] content, int width, int height);

    void setDataSource(Context context, Uri uri, Map<String, String> headers) throws IOException;

    int getTrackCount();

    void unselectTrack(int index);

    MediaFormat getTrackFormat(int index);

    void selectTrack(int index);

    int getSampleFlags();

    long getSampleTime();

    void advance();
}

File: src/main/java/ca/yyx/hu/extractor/MediaExtractorInterface.kt
package ca.yyx.hu.extractor

import android.content.Context
import android.media.MediaCodec
import android.media.MediaFormat
import android.net.Uri

import java.io.IOException
import java.nio.ByteBuffer

/**
 * @author algavris
 * *
 * @date 30/04/2016.
 */
interface MediaExtractorInterface {
    fun readSampleData(buffer: ByteBuffer, offset: Int): Int

    fun getSampleCryptoInfo(cryptoInfo: MediaCodec.CryptoInfo)

    fun release()

    fun setDataSource(content: ByteArray, width: Int, height: Int)

    @Throws(IOException::class)
    fun setDataSource(context: Context, uri: Uri, headers: Map<String, String>)

    val trackCount: Int

    fun unselectTrack(index: Int)

    fun getTrackFormat(index: Int): MediaFormat

    fun selectTrack(index: Int)

    val sampleFlags: Int

    val sampleTime: Long

    fun advance()
}
